eox_nelp/signals/tests/test_tasks.py

```diff
@@ -76,7 +88,55 @@ def test_dispatch_futurex_progress(self, post_futurex_progress_mock, generate_pr
             course_id=course_id,
             user_has_passing_grade=True,
         )
-        post_futurex_progress_mock.assert_called_with(data)
+        post_futurex_progress_mock.assert_called_with(progress_enrollment_data)
+
+    @override_settings()
+    @patch("eox_nelp.signals.tasks._generate_progress_enrollment_data")
+    @patch("eox_nelp.signals.tasks._post_futurex_progress")
+    def test_not_call_dispatch_logic_setting_not_configured(
+        self, post_futurex_progress_mock, generate_progress_enrollment_data_mock
+    ):
+        """Test `dispatch_futurex_progress` is called but the logic inside not.
+        So `generate_progress_enrollment_data` and `post_futurex_progress` are not called
+        due no setting configured.
+
+        Expected behavior:
+            - generate_progress_enrollment_data is not called due settings.
+            - post_futurex_progress_mock is not called due settings.
+        """
+        user, _ = User.objects.get_or_create(username="vader")
+        course_id = "course-v1:test+Cx105+2022_T4"
+        if hasattr(settings, "ACTIVATE_DISPATCH_FUTUREX_PROGRESS"):
+            delattr(settings, "ACTIVATE_DISPATCH_FUTUREX_PROGRESS")
+
+        dispatch_futurex_progress(course_id, user.id, is_complete=True)
+
+        generate_progress_enrollment_data_mock.assert_not_called()
+        post_futurex_progress_mock.assert_not_called()
+
+    @override_settings()
```

Comment:
why both? this line and 134

Reply:
Because this woth method are from dispatch logic. https://github.com/eduNEXT/eox-nelp/blob/master/eox_nelp/signals/tasks.py#L51-L57

Reply:
I didn't understand your answer, but I removed  the  `@override_settings` decorator from all the tests , and this worked, so my question is the same

Reply:
Researching, the override_Setting decorator is also temporal so doesn't affect the setting after each modification. 
It seems is triggered before and after the block code, 
https://github.com/django/django/blob/b347dc63d5468564cbb7050461565f4f89032664/django/test/utils.py#L455C33-L461

Reply:
So using the decorator after the lock code, the original settings are restored.

Reply:
Great, this answers my question